File: src/main/java/net/kenro/ji/jin/purescript/file/PSFileType.java
package net.kenro.ji.jin.purescript.file;

import com.intellij.openapi.fileTypes.LanguageFileType;
import net.kenro.ji.jin.purescript.icons.PSIcons;
import net.kenro.ji.jin.purescript.PSLanguage;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class PSFileType extends LanguageFileType {

    public static final PSFileType INSTANCE = new PSFileType();
    public static final String DEFAULT_EXTENSION = "purs";

    private PSFileType() {
        super(PSLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Purescript file";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Purescript file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return DEFAULT_EXTENSION;
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return PSIcons.FILE;
    }
}

File: src/main/kotlin/org/purescript/file/PSFileType.kt
package org.purescript.file

import com.intellij.openapi.fileTypes.LanguageFileType
import org.purescript.PSLanguage
import org.purescript.icons.PSIcons
import javax.swing.Icon

object PSFileType : LanguageFileType(PSLanguage) {
    override fun getName(): String {
        return "Purescript file"
    }
    override fun getDescription(): String = "Purescript file"
    override fun getDefaultExtension(): String {
        return "purs"
    }
    override fun getIcon(): Icon = PSIcons.FILE


}